### PR TITLE
Add emitter types

### PR DESCRIPTION
**Pull request by oberbeck**

### CODE REVIEW

**Comment by mathe42** [comment]
Dublicate of #33

**Comment by oberbeck** [comment]
> Dublicate of #33

ouch... I am amazed how similar the solutions are though 🤓

May the better one win ✌️

File: src/classes/emitter.ts [hunk 1]
```diff
@@ -1,41 +1,72 @@
...
+> {
+	#events: {
+		[TKey in keyof TEmitterEvent]?: Listener<this, TEmitterEvent[TKey]>[];
+	} = {};
+
+	on<TKey extends keyof TEmitterEvent>(
+		e: TKey,
+		func: Listener<this, TEmitterEvent[TKey]>,
+	) {
+		if (!Array.isArray(this.#events[e])) {
```
Comment (mathe42) [inline src/classes/emitter.ts:28]: Why this check either it is `undefined` or an Array. (empty arrays doesn't matter)

Comment (oberbeck) [reply src/classes/emitter.ts:28]: Yup, that's what I thought, too. But somehow with this Deno/Node setup it would still complain... hence I added the `!`

**Pull request closed without merging**